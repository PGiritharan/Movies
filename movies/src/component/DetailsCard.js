import React, { useEffect } from 'react';

export const DetailsCard = ({ movie }) => {
    const {title,releaseDate,synopsis}=movie
    useEffect(() => {
        // Get the modal
        var modal = document.getElementById("detailsCard");

        // When the user clicks anywhere outside of the modal, close it
        window.onclick = function (event) {
            if (event.target == modal) {
                modal.style.display = "none";
            }
        }
    })

    return (
        <div className='content-container'>
            <div id="detailsCard" className="modal">
                <div className="modal-content">
                    <div className="modal-header">
                        <div className="box-layout__title">{`${title}-${releaseDate}`}</div>
                    </div>
                    <p>{synopsis}</p>
                </div>
            </div>
        </div>
    )
}

export default DetailsCard;
