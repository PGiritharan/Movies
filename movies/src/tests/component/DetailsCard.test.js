import React from 'react';
import { shallow } from 'enzyme';
import DetailsCard from '../../component/DetailsCard';
import movie from '../fixures/movies';
import toJson from 'enzyme-to-json';

test('should render Poster correctly', () => {
  const wrapper = shallow(<DetailsCard movie={movie}/>);
  expect(toJson(wrapper)).toMatchSnapshot();
});