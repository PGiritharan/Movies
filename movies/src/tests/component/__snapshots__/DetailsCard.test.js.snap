// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Poster correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="modal"
    id="detailsCard"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <div
          className="box-layout__title"
        >
          test moive 1-1983
        </div>
      </div>
      <p>
        Test sysnopsis of the movie
      </p>
    </div>
  </div>
</div>
`;
